run-tests.php

```diff
@@ -1354,6 +1353,16 @@ function run_all_tests_parallel($test_files, $env, $redir_tested) {
 	}
 
 	echo "Spawning workers… ";
+
+	// Windows pipes can't be non-blocking, so use sockets there instead
+	$useSockets = (PHP_OS === "WINNT");
+
+	if ($useSockets) {
+		// IPv6 because nobody uses it, so less chance of collisions, right? ;)
+		$port = (ord('<') << 8) + ord('?');
```

Comment:
No need for a magic constant port. use `stream_socket_server("tcp://[::1]:0")`, and an unused port will be selected. 
```php
$a=stream_socket_server("tcp://[::1]:0");
var_dump(stream_socket_get_name($a, false));
//php shell code:1:
//string(9) "::1:52491"
```

and then we can use ipv4. or at least use it as a fallback.

Reply:
Yes, I thought about using `:0`, I'll try implementing it.

I don't see any value in adding IPv4. Who needs to compile PHP 7.3 on a system with no support for IPv6? It's not like IPv6 *connectivity* is required, just loopback.

Reply:
@hikari-no-yume There's 0 reason to use IPv6. Binding to IPv4 will usually bind both anyway, see http://php.net/manual/en/context.socket.php + `ipv6_v6only`.

Reply:
You know there is almost no IPv6 in China, so sometimes the system administrator will disable all the IPv6 components because they believe it can speed up the system. It's silly but it happens.

Reply:
@hikari-no-yume side note, IPv6 support is ALWAYS available on Windows, but can be disabled in custom builds for some odd reason I do not understand